include/map_textures.h

```diff
@@ -292,23 +290,97 @@ SDL_Texture* get_arrow_down_texture(SDL_Renderer* p_renderer)
+            0, 0, 2, 2, 1, 1, 2, 2, 0, 0,
+            0, 0, 2, 2, 1, 1, 2, 2, 0, 0,
+            0, 2, 2, 2, 1, 1, 2, 2, 2, 0,
+            0, 0, 0, 1, 1, 1, 1, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+            0, 0, 0, 1, 1, 1, 1, 0, 0, 0,
+            0, 0, 1, 0, 1, 1, 0, 1, 0, 0,
+            0, 1, 0, 0, 1, 1, 0, 0, 1, 0,
+            1, 0, 0, 1, 1, 1, 1, 0, 0, 1,
+            0, 0, 1, 0, 1, 1, 0, 1, 0, 0,
+            0, 1, 0, 0, 1, 1, 0, 0, 1, 0,
+            1, 0, 0, 1, 1, 1, 1, 0, 0, 1,
+            0, 0, 1, 1, 1, 1, 1, 1, 0, 0,
+            0, 1, 1, 0, 1, 1, 0, 1, 1, 0,
+            1, 1, 0, 0, 1, 1, 0, 0, 1, 1,
+            1, 0, 0, 0, 0, 0, 0, 0, 0, 1};
+
+    for (int i = 0; i < MAP_TEXTURE_ARROW_WIDTH * MAP_TEXTURE_ARROW_HEIGHT; i++)
+    {
+      if (pixels[i] == 2)
+      {
+          pixels[i] = 0xf51da9ff;
+      }
+      else if (pixels[i] == 1)
+      {
+          pixels[i] = 0xefc7ffff;
+      }
+    }
+
+    SDL_LockSurface(p_surface);
+    SDL_memcpy(p_surface->pixels, pixels, p_surface->h * p_surface->pitch);
+    SDL_UnlockSurface(p_surface);
+
+    SDL_Texture* p_texture = SDL_CreateTextureFromSurface(p_renderer, p_surface);
+
+    return p_texture;
+}
+
+
+static inline SDL_Texture* get_barrier_texture(SDL_Renderer* p_renderer, int length)
 {
     int width = length;
-    int height = 3;
 
     SDL_Surface* p_surface = SDL_CreateRGBSurface(SDL_SWSURFACE,
                                                   width,   //width,
-                                                  height,  //height,
+                                                  MAP_TEXTURE_BARRIER_HEIGHT,  //height,
                                                   32,    //depth,
                                                   0xff000000,  // Rmask,
                                                   0x00ff0000,  // Gmask,
                                                   0x0000ff00,  // Bmask,
                                                   0x000000ff); //Amask
 
-    Uint32 pixels[height * width];
+    Uint32 pixels[MAP_TEXTURE_BARRIER_HEIGHT * width];
```

Comment:
An error occurred with this code line, indicating something wrong with the width here.